### PR TITLE
Add himl-config-merger script

### PR DESCRIPTION
**Pull request by costimuraru**
The `himl-config-merger` script, contains logic of merging a hierarchical config directory and creating the end result YAML files.

```sh
himl-config-merger examples/complex --output-dir merged_output --levels env region cluster --leaf-directories cluster
```

Input example:
```
> tree examples/complex
examples/complex
├── default.yaml
├── env=dev
│   ├── env.yaml
│   ├── region=us-east-1
│   │   ├── cluster=cluster1
│   │   │   └── cluster.yaml
│   │   ├── cluster=cluster2
│   │   │   └── cluster.yaml
│   │   └── region.yaml
│   └── region=us-west-2
│       ├── cluster=cluster1
│       │   └── cluster.yaml
│       └── region.yaml
└── env=prod
    ├── env.yaml
    └── region=eu-west-2
        ├── cluster=ireland1
        │   └── cluster.yaml
        └── region.yaml
```

Output:
```
merged_output
├── dev
│   ├── us-east-1
│   │   ├── cluster1.yaml
│   │   └── cluster2.yaml
│   └── us-west-2
│       └── cluster1.yaml
└── prod
    └── eu-west-2
        └── ireland1.yaml
```

### CODE REVIEW

**Review by amuraru — APPROVED** [review]

File: examples/complex/env=dev/region=us-east-1/region.yaml [hunk 1]
```diff
@@ -1,2 +1 @@
-region:
-  location: us-east-1
\ No newline at end of file
+region: us-east-1
```
Comment (ecojan) [inline examples/complex/env=dev/region=us-east-1/region.yaml:1]: new line missing

File: himl/config_merger.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    # load the !include tag
+    Loader.add_constructor('!include', Loader.include)
+
+    # override the Yaml FullLoader with our custom loader
+    yaml.FullLoader = Loader
+
+    # extract the list of absolute paths for leaf directories
+    dirs = get_leaf_directories(opts.path, opts.leaf_directories)
+
+    # merge the configs for using HIML
```
Comment (ecojan) [inline himl/config_merger.py:167]: remove the for in the comment

File: examples/complex/env=dev/region=us-west-2/region.yaml [hunk 3]
```diff
@@ -1,2 +1 @@
-region:
-  location: us-west-2
\ No newline at end of file
+region: us-west-2
```
Comment (ecojan) [inline examples/complex/env=dev/region=us-west-2/region.yaml:1]: new line missing

**Review by ecojan — COMMENTED** [review]
Few minor comments and can be approved 👍  thank you for migrating this here!

One quick question. The original `configMerger` also added a custom YAML tag `!include` which I see was included in the `config_merger.py` script as well. I think we should add a small documentation about this functionality as well as add an example.

**Comment by costimuraru** [comment]
> Few minor comments and can be approved 👍 thank you for migrating this here!
> 
> One quick question. The original `configMerger` also added a custom YAML tag `!include` which I see was included in the `config_merger.py` script as well. I think we should add a small documentation about this functionality as well as add an example.


Good point. Migrated the documentation on that part as well. Have a look.

**Review by ecojan — APPROVED** [review]

**Pull request merged**